### PR TITLE
Clean up pivot selector from peers and unit test

### PR DESCRIPTION
**Pull request by Matilda-Clerke**
## PR description
Simplify the code in PivotSelectorFromPeers.

### CODE REVIEW

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeersTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class PivotSelectorFromPeersTest {
+  private EthContext ethContext;
+  private EthPeers ethPeers;
```
Comment (macfarla) [inline ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeersTest.java:34]:
```suggestion
  @Mock private EthPeers ethPeers;
```

Comment (macfarla) [reply ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeersTest.java:34]: if you use the annotation you don't need to instantiate the mock in the before() method - it's a bit shorter and clearer

Comment (macfarla) [reply ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeersTest.java:34]: you might need to add this for the class
@ExtendWith(MockitoExtension.class)

Comment (Matilda-Clerke) [reply ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeersTest.java]: Thanks, used to do something similar with junit 4 and couldn't get it to work with junit 5

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeers.java [hunk 2]
```diff
@@ -86,41 +87,24 @@ protected Optional<FastSyncState> fromBestPeer(final EthPeer peer) {
...
-  private long countPeersThatCanDeterminePivotBlock() {
-    return ethContext
-        .getEthPeers()
-        .streamAvailablePeers()
-        .filter(this::canPeerDeterminePivotBlock)
-        .count();
-  }
-
-  private boolean canPeerDeterminePivotBlock(final EthPeer peer) {
-    LOG.debug(
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeers.java:118]: we're losing this log line, is that intentional?

**Review by macfarla — APPROVED** [review]
just a Q about logging

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeers.java (hunk 2 above)
Comment (Matilda-Clerke) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeers.java:118]: It's not a particularly good log as it doesn't include any context and is likely to end up listing peers. I can look to add something similar if you think it's worthwhile

Comment (macfarla) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotSelectorFromPeers.java:118]: all good just checking it won't be missed

**Pull request merged**